streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1347,19 +1360,30 @@ private <S extends StateStore> InternalTopicConfig createChangelogTopicConfig(fi
     }
 
     public boolean hasOffsetResetOverrides() {
-        return !(earliestResetTopics.isEmpty() && earliestResetPatterns.isEmpty()
-            && latestResetTopics.isEmpty() && latestResetPatterns.isEmpty());
+        return noneResetTopics.size() + noneResetPatterns.size()
```

Comment:
This is some checkstyle nonsense -- build failed with "too complex condition" warning/error after I added `[none|duration]Reset[Topic|Pattern].isEmpty()`. -- Instead of adding a suppression for it, I just rewrote it to make checkstyle happy.

Reply:
🤷‍♀️